src/anomalib/data/datamodules/depth/adam_3d.py

```diff
@@ -0,0 +1,136 @@
+            Defaults to ``None``.
+        augmentations (Transform | None): General augmentations to apply if stage-specific
+            augmentations are not provided.
+        test_split_mode (TestSplitMode | str): Method to create test set.
+            Defaults to ``TestSplitMode.FROM_DIR``.
+        test_split_ratio (float): Fraction of data to use for testing.
+            Defaults to ``0.2``.
+        val_split_mode (ValSplitMode | str): Method to create validation set.
+            Defaults to ``ValSplitMode.SAME_AS_TEST``.
+        val_split_ratio (float): Fraction of data to use for validation.
+            Defaults to ``0.5``.
+        seed (int | None, optional): Random seed for reproducibility.
+            Defaults to ``None``.
+    """
+
+    def __init__(
+        self,
+        root: Path | str = "./datasets/ADAM3D",
+        category: str = "1m1",
+        train_batch_size: int = 32,
+        eval_batch_size: int = 32,
+        num_workers: int = 8,
+        train_augmentations: Transform | None = None,
+        val_augmentations: Transform | None = None,
+        test_augmentations: Transform | None = None,
+        augmentations: Transform | None = None,
+        test_split_mode: TestSplitMode | str = TestSplitMode.FROM_DIR,
+        test_split_ratio: float = 0.2,
+        val_split_mode: ValSplitMode | str = ValSplitMode.SAME_AS_TEST,
+        val_split_ratio: float = 0.5,
+        seed: int | None = None,
+    ) -> None:
+        super().__init__(
+            train_batch_size=train_batch_size,
+            eval_batch_size=eval_batch_size,
+            num_workers=num_workers,
+            train_augmentations=train_augmentations,
+            val_augmentations=val_augmentations,
+            test_augmentations=test_augmentations,
+            augmentations=augmentations,
+            test_split_mode=test_split_mode,
+            test_split_ratio=test_split_ratio,
+            val_split_mode=val_split_mode,
+            val_split_ratio=val_split_ratio,
+            seed=seed,
+        )
+
+        self.root = Path(root)
+        self.category = category
+
+    def _setup(self, _stage: str | None = None) -> None:
+        """Set up the datasets.
+
+        Args:
+            _stage (str | None, optional): Stage of setup. Not used.
+                Defaults to ``None``.
+        """
+        self.train_data = ADAM3DDataset(
+            split=Split.TRAIN,
+            root=self.root,
+            category=self.category,
+        )
+        self.test_data = ADAM3DDataset(
+            split=Split.TEST,
+            root=self.root,
+            category=self.category,
+        )
+
+    def prepare_data(self) -> None:
+        """Download the dataset if not available."""
+        if (self.root / self.category).is_dir():
+            logger.info("Found the dataset.")
+        else:
+            download_and_extract(self.root, DOWNLOAD_INFO)
```

Comment:
Do you have the pre-commit hooks installed? If not this might help with the file ending
```bash
uv pip install pre-commit
pre-commit run --all-files
```